cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -402,30 +404,46 @@ public boolean apply(RunRecordMeta input) {
     }
   }
 
-  private List<RunRecordMeta> getSuspendedRuns(ProgramId programId, long startTime, long endTime, int limit,
-                                               @Nullable Predicate<RunRecordMeta> filter) {
+  private  Map<ProgramRunId, RunRecordMeta> getSuspendedRuns(ProgramId programId, long startTime, long endTime,
+                                                             int limit, @Nullable Predicate<RunRecordMeta> filter) {
     return getNonCompleteRuns(programId, TYPE_RUN_RECORD_SUSPENDED, startTime, endTime, limit, filter);
   }
 
-  private List<RunRecordMeta> getActiveRuns(ProgramId programId, final long startTime, final long endTime, int limit,
-                                            @Nullable Predicate<RunRecordMeta> filter) {
+  private  Map<ProgramRunId, RunRecordMeta> getActiveRuns(ProgramId programId, final long startTime, final long endTime,
+                                                          int limit, @Nullable Predicate<RunRecordMeta> filter) {
     return getNonCompleteRuns(programId, TYPE_RUN_RECORD_STARTED, startTime, endTime, limit, filter);
   }
 
-  private List<RunRecordMeta> getNonCompleteRuns(ProgramId programId, String recordType,
+  private Map<ProgramRunId, RunRecordMeta> getNonCompleteRuns(ProgramId programId, String recordType,
                                                  final long startTime, final long endTime, int limit,
                                                  Predicate<RunRecordMeta> filter) {
     MDSKey activeKey = getProgramKeyBuilder(recordType, programId).build();
 
-    return list(activeKey, null, RunRecordMeta.class, limit, andPredicate(new Predicate<RunRecordMeta>() {
-                  @Override
-                  public boolean apply(RunRecordMeta input) {
-                    return input.getStartTs() >= startTime && input.getStartTs() < endTime;
-                  }
-                }, filter));
+    Map<MDSKey, RunRecordMeta> runRecordMap = listKV(activeKey, null, RunRecordMeta.class, limit,
+                                                     andPredicate(new Predicate<RunRecordMeta>() {
+      @Override
+      public boolean apply(RunRecordMeta input) {
+        return input.getStartTs() >= startTime && input.getStartTs() < endTime;
+      }
+    }, filter));
+
+    return (getProgramRunIdMap(runRecordMap));
+  }
+
+  /** Converts MDSkeys in the map to ProgramIDs
+   *
+   * @param keymap map with keys as MDSkeys
+   * @return map with keys as program IDs
+   */
+  private Map<ProgramRunId, RunRecordMeta> getProgramRunIdMap(Map<MDSKey, RunRecordMeta> keymap) {
+    Map<ProgramRunId, RunRecordMeta> programRunIdMap = new HashMap<>();
```

Comment:
This method modifies the values returned by HBase, right? Hence this too needs to preserve order. Also, it would be good to add this to the method javadoc so it is part of the contract.